components/opponent/wikis/trackmania/opponent_display_custom.lua

```diff
@@ -0,0 +1,198 @@
+---
+-- @Liquipedia
+-- wiki=trackmania
+-- page=Module:OpponentDisplay/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local DisplayUtil = require('Module:DisplayUtil')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+
+local Opponent = Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+local PlayerDisplay = Lua.import('Module:Player/Display', {requireDevIfEnabled = true})
+
+local OpponentDisplayCustom = Table.deepCopy(OpponentDisplay)
+
+local SCORE_STATUS = 'S'
+local NO_SCORE = -1
+local ZERO_SCORE = 0
+
+--[[
+Display component for an opponent entry appearing in a bracket match.
+]]
+OpponentDisplayCustom.BracketOpponentEntry = Class.new(
+	function(self, opponent, options)
+		self.content = mw.html.create('div'):addClass('brkts-opponent-entry-left')
+
+		if opponent.type == Opponent.team then
+			self:createTeam(opponent.template or 'tbd', options)
+		elseif opponent.type == Opponent.solo or opponent.type == Opponent.duo then
+			self:createPlayers(opponent)
+		elseif opponent.type == Opponent.literal then
+			self:createLiteral(opponent.name or '')
+		end
+
+		self.root = mw.html.create('div'):addClass('brkts-opponent-entry')
+			:node(self.content)
+	end
+)
+
+OpponentDisplayCustom.BracketOpponentEntry.createTeam = OpponentDisplay.BracketOpponentEntry.createTeam
+
+function OpponentDisplayCustom.BracketOpponentEntry:createPlayers(opponent)
+	local players = opponent.players
+	if #players == 1 then
+		local playerNode = PlayerDisplay.BlockPlayer({
+			player = players[1],
+			overflow = 'ellipsis',
+		})
+		self.content:node(playerNode)
+	else
+		local playersNode = OpponentDisplayCustom.PlayerInlineOpponent{
+			opponent = opponent
+		}
+
+		self.content:node(playersNode)
+	end
+end
+
+OpponentDisplayCustom.BracketOpponentEntry.createLiteral = OpponentDisplay.BracketOpponentEntry.createLiteral
```

Comment:
Shouldn't be needed

Reply:
since the `OpponentDisplayCustom.BracketOpponentEntry` class is defined above the one from the commons module gets overwritten
including the sub tables
hence i think it is needed

Reply:
Ah right...Let's move them both of them to right after the init function then

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2262/commits/d41cb109c0ba4f4355e4fa10da9c51c1394d6913